### PR TITLE
feat: add liquidity tabs2

### PR DESCRIPTION
**Pull request by agualis**

### CODE REVIEW

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 1]
```diff
@@ -1,66 +1,141 @@
...
-            </Button>
+          return option?.tabTooltipLabel ? (
+            <Box flex="1" key={`button-group-option-${option.value}`}>
+              <Popover trigger="hover">
+                <PopoverTrigger>
+                  <Box _hover={{ opacity: 0.75 }} transition="opacity 0.2s var(--ease-out-cubic)">
+                    <GroupOptionButton isActive={isActive} option={option} {...props} />
+                  </Box>
+                </PopoverTrigger>
+                {option?.tabTooltipLabel && (
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:58]: when the label is set correctly this is not needed

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 2]
```diff
@@ -1,66 +1,141 @@
...
+                    </Text>
+                  </PopoverContent>
+                )}
+              </Popover>
+            </Box>
+          ) : (
+            <Box flex="1" key={`button-group-option-${option.value}`}>
+              <GroupOptionButton
+                isActive={isActive}
+                key={`button-group-option-${option.value}`}
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:71]: key is probably not needed here

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 3]
```diff
@@ -1,66 +1,141 @@
...
+  groupId,
+  hasLargeTextLabel,
+  onChange,
+}: { option: ButtonGroupOption; isActive: boolean } & Props) {
+  return (
+    <Button
+      bg="transparent"
+      id={`button-group-${option.value}`}
+      isDisabled={option.disabled}
+      key={`button-group-option-${option.value}`}
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:97]: i think this key is also probably not needed

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 4]
```diff
@@ -1,66 +1,141 @@
...
+      <Box fontSize={hasLargeTextLabel ? 'lg' : undefined} position="relative" zIndex="8">
+        {option.label}
+      </Box>
+    </Button>
+  )
+}
+
+function IconPopover({ option }: { option: ButtonGroupOption }) {
+  if (option.tabTooltipLabel) return null
+  if (!option.iconTooltipLabel) return null
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:126]: `if (option.tabTooltipLabel || !option.iconTooltipLabel) return null`

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 5]
```diff
@@ -1,66 +1,141 @@
...
+  return (
+    <Popover key={option.iconTooltipLabel} placement="top" trigger="hover">
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:128]: i think the key here can be removed

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx [hunk 6]
```diff
@@ -1,66 +1,141 @@
...
+      <PopoverTrigger>
+        <Icon as={Info} />
+      </PopoverTrigger>
+      {option?.iconTooltipLabel && (
```
Comment (groninge01) [inline packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:132]: the condition here is not needed i think

File: packages/lib/shared/components/errors/UnbalancedAddError.tsx [hunk 7]
```diff
@@ -1,30 +1,94 @@
...
+
+export function getErrorLabels(
+  isProportionalSupported: boolean,
+  error?: Error | null
+): UnbalancedErrorLabels | undefined {
+  let errorTitle = 'Token amounts error'
+  let errorMessage = 'Unexpected error. Please ask for support'
+  let proportionalLabel = 'Please use proportional mode.'
+
+  if (!error) return
```
Comment (groninge01) [inline packages/lib/shared/components/errors/UnbalancedAddError.tsx:68]: move this to the top of the function

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx (hunk 1 above)
Comment (agualis) [reply packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:58]: Removed

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx (hunk 2 above)
Comment (agualis) [reply packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:71]: If we had multiple options with no `tabTooltipLabel`, we would need that key do differentiate the different Box components, wouldn't we?

File: packages/lib/shared/components/errors/UnbalancedAddError.tsx [hunk 8]
```diff
@@ -1,30 +1,94 @@
...
+  isInvariantRatioPIErrorMessage,
+  isUnbalancedAddError,
+  isUnbalancedAddErrorMessage,
+} from '../../utils/error-filters'
 import { BalAlertLink } from '../alerts/BalAlertLink'
 import { ErrorAlert } from './ErrorAlert'
+import { useAddLiquidity } from '@repo/lib/modules/pool/actions/add-liquidity/AddLiquidityProvider'
 
 type Props = AlertProps & {
   error?: Error | null
```
Comment (groninge01) [inline packages/lib/shared/components/errors/UnbalancedAddError.tsx:16]: why is `error` optional?

File: packages/lib/shared/components/errors/UnbalancedAddError.tsx [hunk 9]
```diff
@@ -1,30 +1,94 @@
...
+
+export type UnbalancedErrorLabels = {
+  errorTitle: string
+  errorMessage: string
+  proportionalLabel: string
+}
+
+export function getErrorLabels(
+  isProportionalSupported: boolean,
+  error?: Error | null
```
Comment (groninge01) [inline packages/lib/shared/components/errors/UnbalancedAddError.tsx:62]: why is `error` optional?

File: packages/lib/shared/utils/error-filters.ts [hunk 10]
```diff
@@ -57,7 +57,7 @@ export function isUnbalancedAddError(error?: Error | null): boolean {
   return false
 }
 
-export function isUnbalancedAddErrorMessage(error: Error | null): boolean {
+export function isUnbalancedAddErrorMessage(error?: Error | null): boolean {
```
Comment (groninge01) [inline packages/lib/shared/utils/error-filters.ts:60]: why is `error` optional?

**Review by groninge01 — APPROVED** [review]

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx (hunk 2 above)
Comment (groninge01) [reply packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:71]: the key is already on the parent

File: packages/lib/shared/utils/error-filters.ts (hunk 10 above)
Comment (agualis) [reply packages/lib/shared/utils/error-filters.ts:60]: We call it with` isUnbalancedAddErrorMessage(priceImpactQuery.error)` where error can be undefined.

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx (hunk 6 above)
Comment (agualis) [reply packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:132]: Removed

File: packages/lib/shared/components/btns/button-group/ButtonGroup.tsx (hunk 5 above)
Comment (agualis) [reply packages/lib/shared/components/btns/button-group/ButtonGroup.tsx:128]: Removed

File: packages/lib/shared/utils/error-filters.ts (hunk 10 above)
Comment (groninge01) [reply packages/lib/shared/utils/error-filters.ts:60]: looks like it's defined or null everywhere

File: packages/lib/shared/components/errors/UnbalancedAddError.tsx (hunk 9 above)
Comment (agualis) [reply packages/lib/shared/components/errors/UnbalancedAddError.tsx:62]: Refactored

**Pull request merged**